src/extension.ts

```diff
@@ -94,8 +97,8 @@ export function activate(context: vscode.ExtensionContext) {
     const documentInfo = {
       bucket: pathParts[0],
       scope: pathParts[1],
-      collection: pathParts[2],
-      name: pathParts[3].substring(0, pathParts[3].indexOf(".json"))
+      collection: pathParts[3],
```

Comment:
Should this be a utility function that you ask for the name of, and it figures out how to split it apart from the pathParts?  This just seems like a log of magic ints that are looking at indexes without understanding why you need part 3 instead of part 2.  At least we should have a breakdown of what this is indexes-wise (what is 0, 1, 2, 3, and 4).  This would make maintaining this code a lot easier than guesswork.

Reply:
This has been used at many places in code so I made it to a single functionality function so that if indexes change I can just refactor this function. We are storing the path from bucket to a particular document. 
I will put the comments in code and signify what's there in part 3.